substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -243,15 +243,35 @@ pub fn migrate_farms<T: Config>() -> frame_support::weights::Weight {
             }
         }
 
-        let replaced_farm_name = farm::replace_farm_name_spaces_with_underscores(&farm.name);
-        let name = match <T as Config>::FarmName::try_from(replaced_farm_name) {
+        let mut farm_name = farm.name;
+        let mut truncated = false;
+        if farm_name.len() > 40 {
+            info!("farm name length exceeded, removing from farm id by name map");
+            FarmIdByName::<T>::remove(&farm_name);
+            farm_name.truncate(40);
+            truncated = true;
+        }
```

Comment:
mutability for truncated can be removed, either by setting truncated to the result of the if check `let truncated = farn_name.len() > 40;` and then checking on `if truncated { ... }`, or by assigning truncated as the result of the `if` block (and introducing an else block), i.e. `let truncated = if farm_name.len() > 40 { ...; true } else { false };`.